### DISCUSSION TITLE
User role automatically uploaded

### DISCUSSION THREAD

**Discussion by themtaysw** [VirtusLab-Open-Source/strapi-plugin-comments#193] [Q&A] [answered]
Hey, I just quickly want to ask a question. How can i modify the plugin a bit so that i can include more fields to be uploaded within the user object? I want the user role to be uploaded, his verified status and also his avatar url... How can i do this? PLEASE????

thanks in advance :)

**Comment by cyp3rius** [comment, marked as answer]
Hello @themtaysw,

You should follow the official Strapi docs in terms of [Plugin Extensions](https://docs.strapi.io/dev-docs/plugins-extension). In the plugin you must play with the service extension.

Anyway for what you would like to achieve the [Populate docs](https://docs.strapi.io/dev-docs/api/rest/populate-select#population) is proper and by adding for example:
- `?populate[author][populate][0]=avatar`
- `?populate[author][populate][0]=role`
You can get what you expect. We're supporting default [populate query](https://github.com/VirtusLab-Open-Source/strapi-plugin-comments#get-comments-flat-structure) of Strapi.
